source/common/upstream/upstream_impl.cc

```diff
@@ -867,12 +866,11 @@ Network::TransportSocketFactoryPtr createTransportSocketFactory(
   auto transport_socket = config.transport_socket();
   if (!config.has_transport_socket()) {
     if (config.has_hidden_envoy_deprecated_tls_context()) {
-      transport_socket.set_name(Extensions::TransportSockets::TransportSocketNames::get().Tls);
+      transport_socket.set_name("envoy.transport_sockets.tls");
```

Comment:
I think the changes to adding the names inline in `name()` methods in source/extensions make sense to me. Here and in `source/server`, references to tls/raw_buffer names seem like they should refer to a common string, but it doesn't seem like there's a good place for that unless it exists in `common/network` somewhere.
@mattklein123 @htuch do you have ideas or should this just stay as is? 
If it should stay as is, PR LGTM.